File: mario.c
#include <cs50.h>
#include <stdio.h>

void pyramid(int n);
void left_pyr(int row, int height);
void right_pyr(int row);

int main(void)
{
    int height;

    do {
        printf("Height: ");
        height = get_int();
    } while (height < 0 || height > 23);

    pyramid(height);
}

void pyramid(int n)
{
    for (int i = 0; i < n; i++) {
        left_pyr(i, n);
        printf("  ");
        right_pyr(i);
        printf("\n");
    }
}

void left_pyr(int row, int height)
{
    for (int i = 0; i < height; i++) {
        if (i + row + 1 < height) {
            printf(" ");
        }
        else {
            printf("#");
        }
    }
}

void right_pyr(int row)
{
    for (int i = 0; i <= row; i++) {
        printf("#");
    }
}

File: mario.py
import cs50

def main():
    while True:
        height = cs50.get_int("Height: ")
        if height >= 0 and height <= 23:
            break
    pyramid(height)    
        
def pyramid(height):
    for i in range (height):
        left_side_row(i, height)
        print(" ", end = "")
        right_side_row(i)

def left_side_row(row, height):
    for i in range(height):
        if i + row + 1 < height:
            print(" ", end = "")
        else:
            print("#", end = "")
    

def right_side_row(row):
    for i in range(row + 1):
        print("#", end = "")
    print("\n", end = "")    
    

if __name__ == "__main__":
    main()